nutils/numeric.py

```diff
@@ -591,4 +591,35 @@ def levicivita(n: int, dtype=float):
     result[tuple(I)] = 1 - 2*(nperms % 2)
     return result
 
+
+def sinc(x, n=0):
+    '''
+    Evaluates the n-th derivative of the unnormalized sinc function:
+
+        sinc(x) = sin(x) / x
+
+    Derivatives are evaluated using the following recurrence relation, which is
+    easily verified to hold for all i > 0:
+
+        sinc'i(x) = (sin'i(x) - i sinc'i-1(x)) / x
+
+    The values at x=0 follow from the Taylor series sinc(x) = Σ_i cos(½πi) x^i / (i+1)!
+
+        sinc'n(0) = cos(½πn) / (n+1)
+    '''
+
+    f = numpy.sinc(x / numpy.pi) # Numpy's implementation is normalized to π
+    # NOTE: alternatively we could initialize f with zeros and iterate from i=0
+    # below, but we prefer to benefit from the numerical stability that Numpy's
+    # implementation provides
+    for i in range(1, n+1):
+        f *= -i
+        f += numpy.sin(x + i * numpy.pi / 2) # derivative corresponds to ½π shift
+        with numpy.errstate(all='ignore'): # results for x=0 are overwitten below
+            f /= x
```

Comment:
The derivative is not stable in the neighborhood of 0:

```python
>>> nutils.numeric.sinc(numpy.array([1e-6, 1e-7, 1e-8, 1e-9, 0]), n=2)
array([-0.33320005, -0.33386618, -0.99999998, -0.99999996, -0.33333333])
```

Reply:
fixed